### PR TITLE
migrate remaining unit tests for 2.0

### PR DESCRIPTION
**Pull request by rymorale**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingExtension.java [hunk 1]
```diff
@@ -459,7 +454,7 @@ private static Map<String, Object> getXdmData(final String eventType, final Stri
         final Map<String, Object> trackingMap = new HashMap<>();
         final Map<String, Object> customActionMap = new HashMap<>();
 
-        if (actionId != null) {
+        if (!StringUtils.isNullOrEmpty(actionId)) {
```
Comment (sbenedicadb) [inline code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingExtension.java:457]: is empty string valid?

**Review by sbenedicadb — APPROVED** [review]
looks good.  just one question about the change to disallow empty string which is a change in behavior potentially for reporting

File: code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingExtension.java (hunk 1 above)
Comment (rymorale) [reply code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingExtension.java:457]: double checked the ios implementation and an empty string is valid. I'll revert this change.

Comment (rymorale) [reply code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingExtension.java:457]: looking into this further, imo an empty string is not valid. the action id is extracted in this line:
```java
final String actionId = eventData.optString(MessagingConstants.EventDataKeys.Messaging.TRACK_INFO_KEY_ACTION_ID, null);
```
with the check above previously checking for null string only. i think updating it to using the StringUtils method is not necessary as the value will always be a valid string or null.

**Pull request merged**